### PR TITLE
[ws-manager] Only extract secrets when FF is set

### PR DESCRIPTION
**Pull request by csweichel**
## Description
Otherwise content init fails for private repositories.

## How to test
1. open a workspace on a private repo
2. enable the protected_secrets FF and open a workspace on a private repo

## Release Notes

```release-note
NONE
```

## Werft options:

- [X] /werft with-preview

### CODE REVIEW

**Review by aledbf — APPROVED** [review]

**Comment by geropl** [comment]
Starting to review now...

File: components/ws-manager/pkg/manager/testdata/validateStartReq_invalidType.json [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
-        "servicePrefix": "foobarservice",
+        "service_prefix": "foobarservice",
         "spec": {
             "ideImage": {
                 "webRef": "eu.gcr.io/gitpod-core-dev/buid/theia-ide:someversion"
             },
-            "workspaceImage": "eu.gcr.io/gitpod-dev/workspace-images/ac1c0755007966e4d6e090ea821729ac747d22ac/eu.gcr.io/gitpod-dev/workspace-base-images/github.com/typefox/gitpod:80a7d427a1fcd346d420603d80a31d57cf75a7af",
-            "checkoutLocation": "/",
-            "workspaceLocation": "/",
+            "workspace_image": "eu.gcr.io/gitpod-dev/workspace-images/ac1c0755007966e4d6e090ea821729ac747d22ac/eu.gcr.io/gitpod-dev/workspace-base-images/github.com/typefox/gitpod:80a7d427a1fcd346d420603d80a31d57cf75a7af",
```
Comment (geropl) [inline components/ws-manager/pkg/manager/testdata/validateStartReq_invalidType.json:15]: This looks fishy to me:
 - why the change to `workspace_image`?
 - and why do all other files keep using `workspaceImage`?
:thinking:

**Comment by geropl** [comment]
/werft run

To get a preview env...

**Comment by geropl** [comment]
/werft run with-preview=true

Did not pick up the flag last time...? :shrug:

**Comment by geropl** [comment]
I tested with a private repo and an user env var and [protected_secrets](url) set to:
 - ws-manager from this branch:
   - `true`: :heavy_check_mark: 
   - `false`: :heavy_check_mark: 
 - ws-manager we currently use in prod (ws-clusters and application clusters, `commit-d537b24994f67318dcadb9d29ebc132d36514f8c`):
   - `false`: :heavy_check_mark: 
   - `true`: :boom: 
image

This means:
 - we have to make sure we standup new ws-clusters before using this feature (kind of make sense)
 - during next WebApp deployment we should make sure to update ws-manager as well (the default, and will happen anyway)

**Review by geropl — APPROVED** [review]
Tested with multiple versions, and is safe to roll out either way as long as we keep the feature flag `protectes_secrets` disabled as long as `ws-manager` not rolled out _everywhere_ (WebApp **and** Workspace): https://github.com/gitpod-io/gitpod/pull/11198#issuecomment-1178732067 :+1:

**Comment by atduarte** [comment]
For the others that might have the same question as I did: 
FF = Feature Flag.
